btf-rs/src/btf.rs

```diff
@@ -18,10 +21,20 @@ pub struct Btf {
 }
 
 impl Btf {
-    /// Parse a stand-alone BTF object file and construct a Rust representation for later
-    /// use. Trying to open split BTF files using this function will fail. For split BTF
-    /// files use `Btf::from_split_file()`.
+    /// Parse a stand-alone BTF object file and construct a Rust representation
+    /// for later use. The input file will be mmaped if supported, otherwise the
+    /// BTF data will be cached in memory. See `Btf::mmap_from_file` and
+    /// `Btf::cache_from_file`.
+    ///
+    /// Trying to open split BTF files using this function will fail. For split
+    /// BTF files use `Btf::from_split_file`.
     pub fn from_file<P: AsRef<Path>>(path: P) -> Result<Btf> {
```

Comment:
Following our discussions about naming and APIs.
If we expect to support more "flavors" of the Btf object, I'm wondering if we should keep a single function and add an enum to describe how the Btf object should be built. Maybe make it optional or have a "Default" variant for users that don't really mind? Not super nice API but having N constructors is not nice either, specially if one "hides" the flavor it is being used.

Reply:
Or split construction from building and have something like:
```
let btf = Btf::from_file()::build()
let btf = Btf::with_backend(Backend::Mmap)::from_file()::build()
```
I've seen this pattern more with auxiliary "builder" objects, i.e: `BtfBuilder::new()[.with_backend()].from_file()`